apps/src/templates/teacherNavigation/ElementOrEmptyPage.tsx

```diff
@@ -27,82 +33,155 @@ const ElementOrEmptyPage: React.FC<ElementOrEmptyPageProps> = ({
+      ),
+    },
+    noLessonMaterialsForLegacyCourses: {
+      headline: i18n.lessonMaterialsAreNotAvailable(),
+      descriptionText: i18n.lessonMaterialsLegacyMessage({
+        courseName: courseName,
+      }),
+      imageComponent: (
+        <img src={TeacherDashboardEmptyState} alt={i18n.almostThere()} />
+      ),
+      button: (
+        <Button onClick={navigateToCoursePage} text={i18n.goToCourse()} />
+      ),
+    },
+    noCalendarForLegacyCourses: {
+      headline: i18n.calendarNotAvailable(),
+      descriptionText: i18n.calendarLegacyMessage({courseName: courseName}),
+      imageComponent: (
+        <img src={CalendarNotAvailable} alt={i18n.calendarNotAvailable()} />
+      ),
+      button: null,
+    },
+    noLessonMaterialsForThisLesson: {
+      headline: i18n.lessonMaterialsNone(),
+      descriptionText: null,
+      imageComponent: (
+        <img src={TeacherDashboardEmptyState} alt={i18n.almostThere()} />
+      ),
+      button: null,
+    },
+    noCalendarForThisUnit: {
+      headline: i18n.calendarNotAvailable(),
+      descriptionText: null,
+      imageComponent: (
+        <img src={CalendarNotAvailable} alt={i18n.calendarNotAvailable()} />
+      ),
+      button: null,
+    },
+    noUnitAssignedForCalendarOrLessonMaterials: {
+      headline: i18n.almostThere(),
+      descriptionText: i18n.noUnitAssigned({
+        page: lessonMaterialsOrCalendarPage,
+        courseName: courseName ? courseName : i18n.thisCourse(),
+      }),
+      imageComponent: <img src={NoUnitAssigned} alt={i18n.almostThere()} />,
+      button: (
         <Button onClick={navigateToCoursePage} text={i18n.assignAUnit()} />
-      );
-    }
+      ),
+    },
   };
 
-  const heading =
-    showNoStudents || showNoCurriculumAssigned
-      ? i18n.emptySectionHeadline()
-      : i18n.almostThere();
+  // separate out logic for the calendar page
+  const calendarState = showNoCalendarForLegacyCourses
+    ? EMPTY_STATE.noCalendarForLegacyCourses
+    : showNoUnitAssigned
+    ? EMPTY_STATE.noUnitAssignedForCalendarOrLessonMaterials
+    : EMPTY_STATE.noCalendarForThisUnit;
 
-  const navigate = useNavigate();
+  let currentEmptyState;
 
-  const navigateToCoursePage = () => {
-    navigate(`../${TEACHER_NAVIGATION_PATHS.courseOverview}`, {
-      relative: 'path',
-    });
-  };
+  if (showNoStudents) {
+    currentEmptyState = EMPTY_STATE.noStudents;
+  } else if (showNoCurriculumAssigned) {
+    currentEmptyState = EMPTY_STATE.noCurriculumAssigned;
+  } else if (isOnCalendarPage) {
+    currentEmptyState = calendarState;
+  } else if (showNoLessonMaterialsForLegacyCourses) {
+    currentEmptyState = EMPTY_STATE.noLessonMaterialsForLegacyCourses;
+  } else if (showNoLessonMaterialsForLesson) {
+    currentEmptyState = EMPTY_STATE.noLessonMaterialsForThisLesson;
+  } else {
+    currentEmptyState = EMPTY_STATE.noUnitAssigned;
+  }
+  const {imageComponent, headline, descriptionText, button} = currentEmptyState;
```

Comment:
```suggestion
  const getCurrentEmptyState = () => {
  if (showNoStudents) {
   return EMPTY_STATE.noStudents;
  } else if (showNoCurriculumAssigned) {
   return EMPTY_STATE.noCurriculumAssigned;
   }
   ...
  }
  const {imageComponent, headline, descriptionText, button} = getCurrentEmptyState();
```
What about something like this? Then we don't have to have the modifiable variable and it is a bit clearer how everything groups together.